cybershell/advanced_ai_orchestrator.py

```
@@ -0,0 +1,1074 @@
+                confidence += 0.1
+        
+        for indicator in uncertainty_indicators:
+            if indicator in response_lower:
+                confidence -= 0.1
+        
+        return max(0.1, min(1.0, confidence))
+    
+    def _generate_cache_key(self, model_name: str, prompt: str) -> str:
+        """Generate cache key for model responses"""
+        content = f"{model_name}:{prompt}"
+        return hashlib.md5(content.encode()).hexdigest()
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)